library/dnet.c

```diff
@@ -433,12 +433,13 @@ static int dnet_iterator_server_send_complete(struct dnet_addr *addr, struct dne
 		err = cmd->status;
 
 		/*
-		 * Write CAS error is not a real 'error' in that regard, that we do not remove
-		 * local record, but also do not stop iterator.
-		 *
 		 * Setting @write_error forces iterator to stop.
+		 *
+		 * Stop iteration only if connection was failed on write
+		 * or no space left on remote backend.
 		 */
-		if (err && !send->write_error && (err != -EBADFD))
+		const int critical_error = err && (err == -ENXIO || err == -ENOSPC);
```

Comment:
No, this is wrong.
Server can answer with many errors which we do not know about, we can not proceed with writing data in this case.

EBADFD is special, since it says write has failed because of CAS/timestamp-cas error, which is OK for this iterator.
Actually, we should proceed only if there is such error AND this is the timestamp-cas case.

What errors did you see that you still want to proceed?